Add macos deployment guide (#179)
EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi: deploy_guides/MacOS_TFLite_Guide.md

> # How to Run TensorFlow Lite Models on MacOS

Reviewer: ```suggestion
# How to Run TensorFlow Lite Models on macOS
```

Reviewer: Just being nitpicky about correct capitalization :)

> This guide shows how to set up a TensorFlow Lite Runtime environment on a MacOS device. We'll use [Anaconda](https://www.anaconda.com/) to create a Python environment to install the TFLite Runtime in. It's easy!

Reviewer: ```suggestion
This guide shows how to set up a TensorFlow Lite Runtime environment on a macOS device. We'll use [Anaconda](https://www.anaconda.com/) to create a Python environment to install the TFLite Runtime in. It's easy!
```

## Step 1. Download and Install Anaconda

> First, install [Anaconda](https://www.anaconda.com/), which is a Python environment manager that greatly simplifies Python package management and deployment. Anaconda allows you to create Python virtual environments on your PC without interfering with existing installations of Python. Go to the [Anaconda Downloads page](https://www.anaconda.com/products/distribution) and click the Download button.

Reviewer: ```suggestion
First, install [Anaconda](https://www.anaconda.com/), which is a Python environment manager that greatly simplifies Python package management and deployment. Anaconda allows you to create Python virtual environments on your Mac without interfering with existing installations of Python. Go to the [Anaconda Downloads page](https://www.anaconda.com/products/distribution) and click the Download button.
```

Reviewer: PC usually refers to Windows desktops or laptops, while Mac refers to Macs. You may be too young to have been around during the "I'm a Mac! I'm a PC" commercials :)
https://www.youtube.com/watch?v=qfv6Ah_MVJU

When the download finishes, open the downloaded .exe file and step through the installation wizard. Use the default install options.

## Step 2. Set Up Virtual Environment and Directory
First open up the terminal by opening a Finder window, and press 'Command + Shift + U', and then select Terminal. We'll create a folder called `tflite1` directly in the Home folder (under your username) - you can use any other folder location you like, just make sure to modify the commands below to use the correct file paths. Create the folder and move into it by issuing the following commands in the terminal:

```

> mkdir tflite1

Reviewer: ```suggestion
mkdir ~/tflite1
```

Reviewer: I know that opening a terminal should put you in the Home directory, but I still like specifying the full filepath to where users should create the folder (in this case, `~/tflite1`). Otherwise, inexperienced can get confused about where the folder should be created.

> cd tflite1

Reviewer: ```suggestion
cd ~/tflite1
```

```
conda activate tflite1-env
pip install tensorflow
pip install opencv-python
pip uninstall protobuf
pip install protobuf==3.20.0
```

Download the detection scripts from this repository by issuing:

```
curl https://raw.githubusercontent.com/EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi/master/TFLite_detection_image.py --output TFLite_detection_image.py
curl https://raw.githubusercontent.com/EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi/master/TFLite_detection_video.py --output TFLite_detection_video.py
curl https://raw.githubusercontent.com/EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi/master/TFLite_detection_webcam.py --output TFLite_detection_webcam.py
curl https://raw.githubusercontent.com/EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi/master/TFLite_detection_stream.py --output TFLite_detection_stream.py
```

## Step 3. Move TFLite Model into Directory

> Next, take the custom TFLite model that was trained and downloaded from the Colab notebook and move it into the {username}\tflite1 directory (replacing {username} with your home directory name). If you downloaded it from Colab, it should be in a file called `custom_model_lite.zip`. (If you haven't trained a model yet and just want to test one out, download my "bird, squirrel, raccoon" model by clicking this Dropbox link.) Move that file to the {username}\tflite1 directory. Once it's moved, unzip it using:

Reviewer: ```suggestion
Next, take the custom TFLite model that was trained and downloaded from the Colab notebook and move it into the {username}\tflite1 directory (replacing {username} with your home directory name). If you downloaded it from Colab, it should be in a file called `custom_model_lite.zip`. (If you haven't trained a model yet and just want to test one out, download my "change counter" model by clicking this [Dropbox link](https://www.dropbox.com/scl/fi/4fk8ls8s03c94g6sb3ngo/custom_model_lite.zip?rlkey=zqda21sowk0hrw6i3f2dgbsyy&dl=0).) Move that file to the {username}\tflite1 directory. Once it's moved, unzip it using:
```

Reviewer: Oops, I just realized that I had never added a Dropbox link on my Windows guide! I added a link for the change counter model.
